src/fabian_nemazi_ex/pa02/chutes_simulation.py

```diff
@@ -0,0 +1,179 @@
+        # If ladders input is None we will assign the values to the defaults
+        # follows the same principle all the way
+        if ladders is None:
+            ladders = Board.Default_ladders
+        if chutes is None:
+            chutes = Board.Default_chutes
+        self.goal = Board.Goal if goal is None else goal
+        # We are asking the user for input, therefore we want to check if the
+        # input are valid. Checking for valid ladders and snake format.
+        for start, ending in ladders:
+            # Iterating through a list with tupples
+            if start >= ending:
+                raise ValueError(
+                    "invalid input, ladders must always lead forward")
+        for start, ending in chutes:
+            if start <= ending:
+                raise ValuError(
+                    "invalid input, snakes must always lead downwards")
+        if self.goal <= 0:
+            raise ValueError("the goal must be higher than 0")
+        # Creating a dictionary of the ladders and snakes
+        self.ladders_and_snakes = {start: end
+                                   for start, end in ladders + chutes}
+
+    def goal_reached(self, position):
+        """
+        Method goal_reached() shall return true if it
+        is passed a position at or beyond the goal.
+        """
+        return position >= self.goal
+
+    def position_adjustment(self, position):
+        """
+        handle changes in position due to snakes and ladders.
+        It accepts a position as argument and returns the number of
+        positions the player must move forward, to get to the correct position.
+        If the player is not at the start of a chute or ladder,
+        the method returns 0.
+
+        """
+        new_position = self.ladders_and_snakes.get(position, position)
+
+        return new_position - position
+
+
+class Player:
+    def __init__(self, board):
+        self._board = board
+        self.position = 0
+        self.moves = 0
+
+    def move(self):
+        """
+        The move() method moves the player by implementing a die cast,
+        the following move and, if necessary,
+        a move up a ladder or down a chute.
+        It does not return anything"""
+        self.position += random.randint(1, 6)
+        self.moves += 1
+        self.position += self._board.position_adjustment(self.position)
+
+
+class ResilientPlayer(Player):
+    def __init__(self, board, extra_steps=1):
+        super().__init__(board)
+        self.extra_steps = extra_steps
+
+    def move(self):
+        """
+        the move() method is the same as in the class Player.
+        The little difference is that a resilient player is to
+        take extra steps after a dice is thrown
+        returns nothing
+        """
+        self.position += random.randint(1, 6) + self.extra_steps
+        self.position += self._board.position_adjustment(self.position)
+        self.moves += 1
```

Comment:
This is wrong. The Resilient Player only takes extra steps only if it went down a snake in the previous round.